Monero-Peer-Subnet-Deduplication/code/Rust/src/main.rs

```diff
@@ -33,6 +33,18 @@ use cuprate_p2p_core::{
 };
 use cuprate_wire::{common::PeerSupportFlags, BasicNodeData};
 
+use clap::Parser;
+
+/// A simple tool to find all the reachable nodes on the Monero P2P network. It works by recursively connecting to
+/// every peer we are told about in a peer list message, starting by connecting to the seed nodes.
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+struct Cli {
+   /// Collect peer lists provided by other nodes and write to peer_list.txt
+   #[arg(short, long, action = clap::ArgAction::SetTrue)] // Defines -c/--collect-peer-lists flag
```

Comment:
```suggestion
   #[arg(short, long)] // Defines -c/--collect-peer-lists flag
```
I don't think you need this here